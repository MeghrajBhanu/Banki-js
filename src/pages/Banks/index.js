import image_map from "../../utils/images";
import classes from "./index.module.css"
import SharedHeader from "../../components/SharedHeader/index"
import { useNavigate } from "react-router";
import { useState } from "react";
import { Outlet } from "react-router";
/**
 * Banks Page Shows all the Banks User has Accounts in.
 * @returns {React.FunctionComponent}
 */
const Banks = () => {
  const Navigate=useNavigate();
  const [clicked,setClicked]=useState(false)
  
  const handleImageClick=(e)=>{
    
    setClicked(!clicked)
    Navigate("/login/banks/"+e.target.name);
  }
  return (
    <>
      {clicked && <Outlet />}
      {!clicked && 
      <>
      <SharedHeader />
      <div className="d-flex flex-wrap justify-content-around mt-5">
        {Object.keys(image_map).map((item) => {
          return (
            
              <img
                src={image_map[item]}
                name={item}
                alt="bank logo"
                className=" rounded img-thumbnail"
                id={classes["image"]}
                key={item}
                onClick={handleImageClick}
                style={{ width: "400px" ,margin:"1em" ,background: "rgba(0, 0, 0, 0.4)",transition: "all ease 1s"}}
              ></img>
            
          );
        })}
      </div>
      </>}
    </>
  );
};
export default Banks;
