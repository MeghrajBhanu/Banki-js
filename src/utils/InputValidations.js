export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}