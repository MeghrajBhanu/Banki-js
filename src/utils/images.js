import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;