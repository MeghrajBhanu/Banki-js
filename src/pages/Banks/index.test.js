import { render, screen, waitFor } from "@testing-library/react";
import { onlywrap, renderComponent } from "../../utils/test_helper";
import renderer from "react-test-renderer";
import Banks from ".";

describe("Banks", () => {
  test("snapshot test", () => {
    const tree = renderer.create(onlywrap(<Banks />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
