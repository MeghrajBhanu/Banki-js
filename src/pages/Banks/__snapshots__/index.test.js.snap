// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banks snapshot test 1`] = `
Array [
  <section
    className="shared_header"
  >
    <div
      className="overlay"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col"
            data-wow-delay="0.6s"
            style={
              Object {
                "animationDelay": "0.6s",
                "animationName": "fadeInUp",
                "visibility": "visible",
              }
            }
          >
            <div
              className="fadeInUp"
            >
              <h2
                className="text-center mb-4"
              >
                Welcome 
                user
                 !
              </h2>
              <p
                className="text-center mt-3"
              >
                Here are your 
                 bank accounts linked with 
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="d-flex flex-wrap justify-content-around mt-5"
  >
    <img
      alt="bank logo"
      className=" rounded img-thumbnail"
      id="image"
      name="Bank1"
      onClick={[Function]}
      src="bank.png"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.4)",
          "margin": "1em",
          "transition": "all ease 1s",
          "width": "400px",
        }
      }
    />
    <img
      alt="bank logo"
      className=" rounded img-thumbnail"
      id="image"
      name="Bank2"
      onClick={[Function]}
      src="bank2.png"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.4)",
          "margin": "1em",
          "transition": "all ease 1s",
          "width": "400px",
        }
      }
    />
    <img
      alt="bank logo"
      className=" rounded img-thumbnail"
      id="image"
      name="Bank3"
      onClick={[Function]}
      src="bank3.png"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.4)",
          "margin": "1em",
          "transition": "all ease 1s",
          "width": "400px",
        }
      }
    />
    <img
      alt="bank logo"
      className=" rounded img-thumbnail"
      id="image"
      name="Bank4"
      onClick={[Function]}
      src="bank4.png"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.4)",
          "margin": "1em",
          "transition": "all ease 1s",
          "width": "400px",
        }
      }
    />
  </div>,
]
`;
